### ISSUE TITLE
Notify users when an updated version is available when connecting to MP

### ISSUE THREAD

**Issue by Pentarctagon** [wesnoth/wesnoth#4825]
Some discussion starting [here](https://github.com/wesnoth/wesnoth/pull/3464#issuecomment-596514487), the idea is to notify users when connecting to the MP server if their current version is out of date.

I'd expect this would be most useful for people who download from SourceForge.  Most other distribution methods have automatic updates already, so if the update just hasn't been applied yet for whatever reason, then that's not something the user would need to take any action on themselves.  On the opposite side of things, linux distros may well have the same outdated version downloadable from their repositories for an indefinite amount of time, so the user would get this notification every time they connect to MP without being able to get the latest version.

Given that, there would need to be a "Don't show me this again" option.  Or, since starting with 1.14.10/1.14.11 and 1.15.3 the wesnoth client will tell the server where it was downloaded from, this could be implemented to only show the warning to users who downloaded it from SourceForge and also don't have the latest version.

**Comment by gfgtdf** [comment]
As mentioned in the linked thead this is already implemened, but afaik the server config file still has something like 1.14.4 as latest version.

**Comment by Pentarctagon** [comment]
There would at least need to be something added to not alert people who aren't downloading wesnoth updates manually though, which is most people at this point.

**Comment by soliton-** [comment]
I think a message with the option to not show it again (for that version and perhaps also ever) is simplest. I'm not sure we can safely decide one way or another based on client source since you at least can usually disable updates?

**Comment by gfgtdf** [comment]
> There would at least need to be something added to not alert people who aren't downloading wesnoth updates manually though, which is most people at this point.

i don't see why, everyone should know that their version is outdated, even if their usual way to get i doesn't let them get the latest version. This is even more true If the new version fixed important multiplayer related bugs (like OOS). Also the current messsage shown isn't too annoying since it it just shown in the chat area and doesn't actively have to be clicked away. So i think it toatlly okay to show it to them ever time they log in.

**Comment by Pentarctagon** [comment]
@gfgtdf Ah, alright, I was assuming it was a popup that had to be clicked to be dismissed.

@soliton- My assumption is that if someone intentionally disabled updates, they probably wouldn't re-enable them if we told them there was an update anyway.

**Comment by soliton-** [comment]
@Pentarctagon Right, but then they will get the message all the time that they don't care about? If we keep it a simple server message like the security warning now then I agree it's not that big a deal. However if we want people to read the motd for example it might be bad to condition them to ignore logon messages.

**Comment by Pentarctagon** [comment]
It's a pretty short message now, so probably it's fine.  Otherwise, I think it'd be better to only show it once per version update, rather than the user having to go into the preferences themselves and disable it.

**Comment by Wedge009** [comment]
Is this any different to #3482 or is that one specifically a warning to hosts while this one is for all users who connect to the MP server?

**Comment by Wedge009** [comment]
I thought #4388 was about add-ons. If it's the case that #3482 is also about add-ons, then sounds like they're duplicates.

**Comment by drewdaemon** [comment]
Hi all, what is the status of this issue? Is it still being considered?

**Comment by soliton-** [comment]
Sure, the issue is still open.

I think a logon message like the secure_version check but with a configurable version from the server config would be a reasonable implementation. Should be doable to update the config on each release.
